pyannote/audio/core/pipeline.py

```diff
@@ -135,7 +131,17 @@ def from_pretrained(
         )
         params = config["pipeline"].get("params", {})
         params.setdefault("use_auth_token", use_auth_token)
-        pipeline = Klass(**params)
+
+        try:
+            # If hub repo contains subfolders, load models and pipeline:
+            embedding = Model.from_pretrained(model_id, subfolder="embedding")
+            segmentation = Model.from_pretrained(model_id, subfolder="segmentation")
+            pipeline = Klass(**params)
+            pipeline.embedding = embedding
+            pipeline.segmentation_model = segmentation
```

Comment:
This seem way too specific to `pyannote/speaker-diarization-3.1`.
I'd like to find a better (= more generic) way

We could do something like preprend subfolders by `@model` (or anything that makes sense) to indicate to `Pipeline.from_pretrained` that a model should be loaded from corresponding subfolders. 

```yaml
pipeline:
  name: pyannote.audio.pipelines.SpeakerDiarization
  params:
    clustering: AgglomerativeClustering
    embedding: @model/embedding 
    segmentation: @model/segmentation
```

Similarly, we could use `@pipeline/` to load sub-pipelines, and later `@whatever` if we ever want to add new pyannote stuff (I already have one in mind that I cannot really talk about right now).